### ISSUE TITLE
Call with V8

### ISSUE THREAD

**Issue by hashseed** [nodejs/TSC#454]
We had a few calls to sync between what was the CTC and select people from the V8 team in the past, with good outcomes.

I'd like to propose another one in the second half of January.

Some agenda items I have in mind:
- Update on [GYP/GN](https://docs.google.com/document/d/1gvHuesiuvLzD6X6ONddxXRxhODlOJlxgfoTNZTlKLGA/edit#)
- GCC support in V8
- Visual Studio support in V8
- [Snapshot support in Node.js](https://github.com/nodejs/node/issues/17058#issuecomment-345231793)
- Spectre and Meltdown
- Trace events
- Impact of `async_hooks` performance cliffs
- General `Promise` and `async`/`await` performance considerations for Node 10
- Workers in Node.js
- Memory tooling
- [REPL mode](https://github.com/bmeck/js-repl-goal) for V8 parser
- V8 extras

cc @bmeurer @ajklein @ofrobots @fhinkel @mathiasbynens @natorion

Before I set up a Doodle, I'd like to get a feel for the general opinion on this.

**Comment by bmeurer** [comment]
I'd like to propose the following additions to that list:

- Impact of `async_hooks` performance cliffs
- General `Promise` and `async`/`await` performance considerations for Node 10

cc @mlippautz

**Comment by TimothyGu** [comment]
I think @bmeck would be interested in talking about a special parsing/execution mode for REPL, and perhaps about modules as well if there is anything. For that matter I’d be interested too but I’m not part of the TSC or a regular observer.

**Comment by hashseed** [comment]
I don't think anyone would mind if we invite both @bmeck and @TimothyGu

**Comment by bnoordhuis** [comment]
> Status update on GYP/GN

I have an in-progress cmake branch that I should get merged but it's currently blocked on https://github.com/nodejs/build/issues/1003.

**Comment by hashseed** [comment]
> cmake
Even with the switch to cmake we would need a way for V8 to test against Node.js without having to maintain cmake configurations, so the GN wrapper still makes sense.

**Comment by maclover7** [comment]
Not sure if this is already documented somewhere, but I think having a more written down policy regarding support for VS2015 (is being dropped beginning in v10 IIRC?) would probably be good. IIRC, the change was related in part to things with V8

**Comment by jasnell** [comment]
Count me in. Would love to see this happen again (and becoming a regular thing).

One additional item I would add is a status update on trace event implementation, just so we can keep it on the radar.

**Comment by bmeck** [comment]
Sounds good to me.

**Comment by mhdawson** [comment]
I'd like to attend as well.

**Comment by hashseed** [comment]
Does it make sense to schedule before or after a TSC meeting? I would assume that that would be more convenient for TSC members.

**Comment by mcollina** [comment]
We do have a calendar to schedule TSC meetings when everybody can attend, can we use the same system to schedule? What do you think @Trott?

**Comment by Trott** [comment]
> We do have a calendar to schedule TSC meetings when everybody can attend, can we use the same system to schedule? What do you think @Trott?

It's easy enough if we have the data. The challenge is getting everyone to enter data. Also, the TSC data is always for a Wednesday (UTC). Things get more difficult if you're talking about different days of the week.

For ad hoc meetings like this, I think Doodle might be more appropriate.

**Comment by mhdawson** [comment]
@hashseed . My experience is that its always a pain to try to schedule a time.  What I'd suggest is a doogle ot pick the first meeting and then make agree on a regular time slot.  It's easier for people to remember and less work each time.  Depending on who wants to attend though you might need to agree on a few slots and rotate.

**Comment by mhdawson** [comment]
I can see the rational of trying to piggy back onto the TSC meeting, but it runs long often enough and even if it does not people may be fatigued having it right after an hour meeting.

**Comment by hashseed** [comment]
Okay. Assuming that most are located between Europe and West Coast, I chose european evenings and west coastal mornings for the Doodle: https://doodle.com/poll/d4ubsq5t6qczcdwm

Let's agree on a suitable time before the January 19th. Thanks!

**Comment by Trott** [comment]
I filled out the Doodle, but I'm no doubt the least important person to attend, so use me as a tie-breaker only. (Turns out my availability those days is rather scarce anyway.)

**Comment by aaronleeatali** [comment]
I would like to know the future of workers in node.js. My co-workers and I have worked on this in recent moths to make workers in node.js much lighter.

**Comment by mhdawson** [comment]
@aaronleeatali see https://github.com/nodejs/worker for some past discussion/work on this front.  It would be great if you could comment on existing issues or open a new one there to share your use case and experience with workers.

**Comment by yunong** [comment]
I would be interested in attending.

**Comment by hashseed** [comment]
It's not quite January 19th yet, but I'd like to propose the final choice, based on the Doodle outcome:

**January 31st, 7:15 PM Central European Time, 10:15 AM Pacific Time.**

**Comment by TimothyGu** [comment]
Seems like I'll be one of the only two people to miss the meeting... oh well. I do have a class during that time but I'll see if I can miss it. That's my long-winded way of saying "sounds good to me."

**Comment by aaronleeatali** [comment]
@mhdawson Thanks.

**Comment by aaronleeatali** [comment]
@mhdawson Thanks. 
I am reading this and I hope our work can makes a positive contribution.
We have implemented worker in node.js which is relatively light-weight.  
The design document and the performance data will be published after we finish our job.
 We would be glad  to receive feedback then.

**Comment by hashseed** [comment]
@TimothyGu sorry that the date doesn't suit you. I will take meticulous notes.

**Comment by hashseed** [comment]
Invitation has been sent. Let me know if I should extend it to more folks.

**Comment by mhdawson** [comment]
@hashseed depending on how open it is, it would be great if @mmallick-ca, @jbarz could be invited.  They work with me keeping our V8 ports up to date for s390 and PowerPC

**Comment by hashseed** [comment]
@mhdawson no objections here. I extended the invitation to both.

**Comment by jasnell** [comment]
@hashseed ... can you please include me on that as well? (I haven't seen an invitation come through so just want to make sure)

**Comment by hashseed** [comment]
I thought I already did. In any case I removed and readded you from the calendar event.

**Comment by fhinkel** [comment]
Should we plan on streaming/recording this on YouTube?

**Comment by dshaw** [comment]
@fhinkel Recording this meeting would be great. I would definitely be interested in viewing this, but have been resisting piling on just to observe. Thank you.

**Comment by mhdawson** [comment]
I created a youtube event.  The main limitation is that I think we can only have ~10 people (although I think the limit might be 15 for our account).  https://www.youtube.com/watch?v=Cg1jWTPAIDU is the link for those who want to watch.  I'll post the link for participants a few minutes before the meeting (Same can be done by any "manager" of the Node.js youtube account)

**Comment by TimothyGu** [comment]
Even though I probably will not be there at the meeting, if possible I'd like to add to the agenda to talk about hooks in v8::Module that would allow more efficient deprecation of module exports. Much of the discussion began in https://github.com/nodejs/node/pull/17560, but I'll summarize it below.

Currently, Node.js uses self-hosted CJS files for implementation of core modules. To deprecate an export, we have two ways:

1. If the export is a function, we emit a warning to stderr saying that the function is deprecated and should no longer be used *when the function is first called*.
2. If the export is a regular property, we emit a warning *when the property is first accessed through a getter function on the module*.

While option 1 still works with ES Modules, option 2 does not, since ES Modules' exports are not an object users can manipulate. But even with option 1, the current situation is not ideal, as an exported function may not always be called during development, thus preventing a developer from seeing the deprecation warning.

Of course, we can just change properties to getter/setter function pairs that are each exported. However, with ESM, ideally we would have a chance to introspect a particular ES Module, and to emit a warning every time a deprecated binding gets imported. With the static/declarative nature of ESM syntax, this would be technically possible. Not only would this solve the problem of deprecating properties, but also provide more accurate reports of deprecated functions to users.

Some open questions remain about what we should do with `import *` and `export *`, however.

**Comment by mhdawson** [comment]
I accidentally used the meeting I had set up during the last benchmarking meeting.  So new hangout. Link for those who just want to watch https://www.youtube.com/watch?v=4Fz6Tz9kb_s.  Will post link for participants a few mins before meeting start.s

**Comment by mhdawson** [comment]
Link for participants https://hangouts.google.com/hangouts/_/ytl/emObhwJJLj7HA6hCi7-oPVFpXGNfwukePeqM4RIQVEE=?eid=100598160817214911030

**Comment by bmeurer** [comment]
Would it be possible to reorder the agenda items to have these first?
- Impact of async_hooks performance cliffs
- General Promise and async/await performance considerations for Node 10
- Workers in Node.js
- Memory tooling
- V8 extras

**Comment by ofrobots** [comment]
video call is full 😿

**Comment by cjihrig** [comment]
yes, what @ofrobots said

**Comment by jasnell** [comment]
Ditto. Unable to join. Can we please pick something other than hangouts with higher capacity

**Comment by mhdawson** [comment]
Going to setup a zoom instead

**Comment by mhdawson** [comment]
https://zoom.us/s/157133358

**Comment by mcollina** [comment]
https://zoom.us/j/157133358

**Comment by devsnek** [comment]
is this no longer going to be streamed?

**Comment by Fishrock123** [comment]
The video recording is still processing but I'll post here when it's done.

**Comment by jasnell** [comment]
Good thing you were recording because it looks like the zoom recording wasn't turned on :-)

**Comment by Fishrock123** [comment]
Youtube recording: https://youtu.be/a6YtLIaqMqs

**Comment by Fishrock123** [comment]
@hashseed Think you could get us a link again to the clang presentation slides? I'd like to annotate the recording there.

Unfortunately I was not set up to record the actual video (and didn't really even think about it due to how our regular meetings are). That discussion starts at https://youtu.be/a6YtLIaqMqs?t=47m4s for like 20mins.

**Comment by hashseed** [comment]
@Fishrock123 [here's](https://docs.google.com/presentation/d/1amWFe4-SHLAeG0jfKJCJIS2JBtN1gljgB0T3sAOqEuQ/edit?usp=sharing) the clang presentation.

**Comment by hashseed** [comment]
I created a [summary](https://docs.google.com/document/d/1Iw0i8wIrzoc_32gfAw9aAhluS9fsMs9xWwL7tu46K_g/edit#).
